sky/clouds/runpod.py

```diff
@@ -0,0 +1,283 @@
+""" RunPod Cloud. """
+
+import json
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple
+
+from sky import clouds
+from sky import status_lib
+from sky.clouds import service_catalog
+import sky.skylet.providers.runpod.rp_helper as runpod_api
+
+if typing.TYPE_CHECKING:
+    from sky import resources as resources_lib  # Renaming to avoid shadowing variables.
+
+_CREDENTIAL_FILES = [
+    'credentials.toml',
+]
+
+
+@clouds.CLOUD_REGISTRY.register
+class RunPod(clouds.Cloud):
+    """ RunPod GPU Cloud
+
+    _REPR | The string representation for the RunPod GPU cloud object.
+    """
+    _REPR = 'RunPod'
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.STOP: 'FluffyCloud does not support stopping VMs.',
+        clouds.CloudImplementationFeatures.AUTOSTOP: 'FluffyCloud does not support stopping VMs.',
+        clouds.CloudImplementationFeatures.MULTI_NODE: 'Multi-node is not supported by the FluffyCloud implementation yet.',
+    }
+    _MAX_CLUSTER_NAME_LEN_LIMIT = 120
+    _regions: List[clouds.Region] = []
+
+    @classmethod
+    def _cloud_unsupported_features(cls) -> Dict[clouds.CloudImplementationFeatures, str]:
+        return cls._CLOUD_UNSUPPORTED_FEATURES
+
+    @classmethod
+    def _max_cluster_name_length(cls) -> Optional[int]:
+        return cls._MAX_CLUSTER_NAME_LEN_LIMIT
+
+    @classmethod
+    def regions_with_offering(cls, instance_type: Optional[str],
+                              accelerators: Optional[Dict[str, int]],
+                              use_spot: bool, region: Optional[str],
+                              zone: Optional[str]) -> List[clouds.Region]:
+        assert zone is None, 'RunPod does not support zones.'
+        del accelerators, zone  # unused
+        if use_spot:
+            return []
+        if instance_type is None:
+            # Fall back to default regions
+            regions = cls.regions()
```

Comment:
This is no longer needed as we will always ensure `instance_type` to be provided.